flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamPhysicalJoin.scala

```diff
@@ -116,13 +123,27 @@ class StreamPhysicalJoin(
   }
 
   override def translateToExecNode(): ExecNode[_] = {
+    val stateTtlFromHint = new util.HashMap[JInt, JLong]
+    getHints
+      .filter(hint => StateTtlHint.isStateTtlHint(hint.hintName))
+      .forEach {
+        hint =>
+          hint.kvOptions.forEach(
+            (input, ttl) =>
+              stateTtlFromHint
+                .put(
+                  if (input == FlinkHints.LEFT_INPUT) 0 else 1,
```

Comment:
I have some different ideas here. 
1. In the explain syntax, if the user uses "1d" in the state ttl hint, I think it is easier to understand by displaying "1d" than by displaying "86400000 ms". 
3. The redundancy of the code is only about the "TimeUtils.parseDuration(ttl).toMillis" section, which I think is tolerable. Otherwise, based on the first consideration, we will add "TimeUtils.parseDuration(ttl)" to the explainTerms method, which will also cause some redundancy.

Reply:
> In the explain syntax, if the user uses "1d" in the state ttl hint, I think it is easier to understand by displaying "1d" than by displaying "86400000 ms".

The key lies in whether the explained digest requires a normalized format of everything. If there's no explicit constraint, I'm ok with leaving it unchanged.

> The redundancy of the code is only about the "TimeUtils.parseDuration(ttl).toMillis" section, which I think is tolerable. Otherwise, based on the first consideration, we will add "TimeUtils.parseDuration(ttl)" to explain, which will also cause some redundancy.

Actually, I was suggesting adding a normalizer, like `CapitalizeQueryHintsShutle`, instead of formatting it in the `RelWriterImpl`.

Reply:
Actually, the value of the original state ttl hint written by user is also a standardized format (duration), otherwise an exception will be thrown directly. It seems that there is no need to uniformly convert it to milliseconds.